analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Common/Checks/ByteArrayCheck.cs

```diff
@@ -42,6 +44,16 @@ programPoint.CurrentInstruction switch
                 ? programState.SetConstraint(programState.PeekValue(), ByteArraySymbolicValueConstraint.Constant)
                 : programState;
 
+        private ProgramState ImplicitlyTypedArrayPostProcess(InitializerExpressionSyntax initializerExpression, ProgramState programState) =>
+            initializerExpression.IsKind(SyntaxKind.ArrayInitializerExpression)
+            // when the initializer is in a typed array creation, it is handled by ArrayCreationPostProcess()
+            && !initializerExpression.Parent.IsKind(SyntaxKind.ArrayCreationExpression)
+            && initializerExpression.FirstAncestorOrSelf<VariableDeclarationSyntax>() is VariableDeclarationSyntax variableDeclaration
+            && semanticModel.GetTypeInfo(variableDeclaration.Type).Type.Is(KnownType.System_Byte_Array)
+            && programState.HasValue
+                ? programState.SetConstraint(programState.PeekValue(), ByteArraySymbolicValueConstraint.Constant)
```

Comment:
We're putting `Constant` constraint on every initialization. Let's document FP in case like this (I don't think it's worth inspecting the values to be constant-only in this super-safe approach :) ):
```
            using var rng = new RNGCryptoServiceProvider();
            byte[] initializationVector =
            {
                Rnd(), Rnd(), Rnd(), Rnd(), Rnd(), Rnd(), Rnd(), Rnd(),
                Rnd(), Rnd(), Rnd(), Rnd(), Rnd(), Rnd(), Rnd(), Rnd()
            };

            using (AesCryptoServiceProvider aes = new AesCryptoServiceProvider())
            {
                ICryptoTransform encryptor = aes.CreateEncryptor(aes.Key, initializationVector); // Noncompliant FP, implicitly typed array initialization considers the values as constant
            }

            byte Rnd()
            {
                var buff = new byte[] { 0 };
                rng.GetBytes(buff);
                return buff[0];
            }
```